### PR TITLE
Move examples to their own package

### PR DESCRIPTION
**Pull request by jystic**
Also switched to mafia, cabal new-build was a bit too fiddly / unfamiliar.

### CODE REVIEW

File: hedgehog-example/hedgehog-example.cabal [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+category:
+  Testing
+license:
+  BSD3
+license-file:
+  LICENSE
+cabal-version:
+  >= 1.8
+build-type:
+  Simple
```
Comment (charleso) [inline hedgehog-example/hedgehog-example.cabal]: Woah. Formatting is blowing my mind. :)

**Comment by charleso** [comment]
👍

File: .travis.yml [hunk 2]
```diff
@@ -7,11 +7,10 @@ git:
 cache:
   directories:
     - $HOME/.cabal/packages
-    - $HOME/.cabal/store
+    - $HOME/.mafia
```
Comment (thumphries) [inline .travis.yml]: now this is a mafia cache

Comment (jystic) [inline .travis.yml]: 📦

File: hedgehog-example/hedgehog-example.cabal [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  ghc-options:
+    -Wall -threaded -O2
+
+  hs-source-dirs:
+    test
+
+  build-depends:
+      hedgehog
+    , base                            >= 3          && < 5
```
Comment (charleso) [inline hedgehog-example/hedgehog-example.cabal]: Should the versions be on a new line as well? :-P

Comment (jystic) [inline hedgehog-example/hedgehog-example.cabal]: haha

Comment (jystic) [inline hedgehog-example/hedgehog-example.cabal]: wow, you got me on that one

File: hedgehog-example/hedgehog-example.cabal (hunk 1 above)
Comment (jystic) [inline hedgehog-example/hedgehog-example.cabal]: who knows if it's a good idea, just thought I would try it

**Pull request merged**